src/visitors/CyclomaticComplexityVisitor.java

```diff
@@ -0,0 +1,36 @@
+package visitors;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.expr.BinaryExpr;
+import com.github.javaparser.ast.stmt.IfStmt;
+import com.github.javaparser.ast.visitor.GenericVisitorAdapter;
+
+import java.util.List;
+
+public class CyclomaticComplexityVisitor extends GenericVisitorAdapter<Integer, Void> {
+
+    @Override
+    public Integer visit(ClassOrInterfaceDeclaration n, Void arg) {
+        return countDecisions(n);
+
+    }
+
+    private Integer countDecisions(Node node) {
+        Integer counter = 0;
+
+        // count all if statements
+        List<IfStmt> ifStatements = node.findAll(IfStmt.class);
+        counter += ifStatements.size();
+        for (IfStmt ifStmt : ifStatements) {
+            // add all binary expr within if/else statements that return a Boolean (incl '==')
+            List<BinaryExpr> binaryExprs = ifStmt.getCondition().findAll(BinaryExpr.class);
+            counter += binaryExprs.size();
+            // add else statements
+            if(ifStmt.getElseStmt().isPresent()) {
+                counter++;
+            }
+        }
+        return counter;
+    }
+}
```

Comment:
What is source of the formula for cyclomatic complexity that you are referring to? Wikipedia defines it as
>    M = E − N + 2P,

where

    E = the number of edges of the graph.
    N = the number of nodes of the graph.
    P = the number of connected components.

Reply:
You can ignore this comment.